plugin/trino-hive/src/main/java/io/trino/plugin/hive/aws/athena/PartitionProjectionService.java

```diff
@@ -0,0 +1,332 @@
+import static java.lang.String.format;
+import static java.util.Objects.nonNull;
+import static java.util.function.Function.identity;
+
+public final class PartitionProjectionService
+{
+    private final TypeManager typeManager;
+    private final Map<ProjectionType, ProjectionFactory> projectionFactories;
+
+    @Inject
+    public PartitionProjectionService(TypeManager typeManager, Map<ProjectionType, ProjectionFactory> projectionFactories)
+    {
+        this.typeManager = typeManager;
+        this.projectionFactories = projectionFactories;
+    }
+
+    public Map<String, Object> getPartitionProjectionTrinoTableProperties(Table table)
+    {
+        Map<String, String> metastoreTableProperties = table.getParameters();
+        Map<String, Object> trinoTableProperties = new HashMap<>();
+        Optional.ofNullable(metastoreTableProperties.get(METASTORE_TABLE_PROPERTY_KEY_PROJECTION_ENABLED))
+                .map(Boolean::valueOf)
+                .ifPresent(enabled -> {
+                    trinoTableProperties.put(PARTITION_PROJECTION_ENABLED, enabled);
+                    Optional.ofNullable(metastoreTableProperties.get(METASTORE_TABLE_PROPERTY_KEY_PROJECTION_LOCATION_TEMPLATE))
+                            .ifPresent(property -> trinoTableProperties.put(PARTITION_PROJECTION_LOCATION_TEMPLATE, property));
+                });
+        return trinoTableProperties;
+    }
+
+    public Map<String, Object> getPartitionProjectionTrinoColumnProperties(Table table, String columnName)
+    {
+        Map<String, String> metastoreTableProperties = table.getParameters();
+        if (nonNull(metastoreTableProperties.get(METASTORE_TABLE_PROPERTY_KEY_PROJECTION_ENABLED))) {
+            return rewriteColumnProjectionProperties(metastoreTableProperties, columnName);
+        }
+        return ImmutableMap.of();
+    }
+
+    public Map<String, String> getPartitionProjectionHiveTableProperties(ConnectorTableMetadata tableMetadata)
+    {
+        Map<String, Object> trinoTableProperties = tableMetadata.getProperties();
+        Map<String, String> metastoreTableProperties = new HashMap<>();
+
+        // Handle Table Properties
+        Optional<Boolean> isEnabled = Optional.ofNullable((Boolean) trinoTableProperties.get(PARTITION_PROJECTION_ENABLED));
+        isEnabled.ifPresent(value -> metastoreTableProperties.put(METASTORE_TABLE_PROPERTY_KEY_PROJECTION_ENABLED, value.toString()));
+        Optional.ofNullable(trinoTableProperties.get(PARTITION_PROJECTION_LOCATION_TEMPLATE))
+                .ifPresent(value -> metastoreTableProperties.put(METASTORE_TABLE_PROPERTY_KEY_PROJECTION_LOCATION_TEMPLATE, value.toString()));
+
+        // Handle Column Properties
+        tableMetadata.getColumns().stream()
+                .filter(columnMetadata -> !columnMetadata.getProperties().isEmpty())
+                .forEach(columnMetadata -> {
+                    rewriteProperty(
```

Comment:
nit: extract `columnProperties` and `columnName` to variable in lambda.

Reply:
Applied